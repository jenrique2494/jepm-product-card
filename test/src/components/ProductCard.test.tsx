// import * as ReactDOM from 'react-dom';
import React from 'react';
import renderer  from 'react-test-renderer';
import { product1 } from '../../data/products';
import { ProductCard } from '../../../src/components';

const {act}=renderer;

describe('pruebas en el ProductCard', () => {
  test('debe mostrar el componente correctamente con el productcard personalizado',()=>{
    const wrapper= renderer.create(
        <ProductCard product={product1}>
            {
                ()=>(
                    <h1>
                        'hola mundo'
                    </h1>
                )
            }
        </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot()

  });
  test('debe de incrementar el contador', () => {
    const wrapper= renderer.create(<ProductCard product={product1}>
        {
            ({count,increaseBy})=>(
                <>
                    <h1>Poduct card</h1>
                    <span>{count}</span>
                    <button onClick={()=>increaseBy(1)} ></button>
                </>
            )
        }
    </ProductCard>)
    let tree =wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(()=>{
        (tree as any).children[2].props.onClick();
    })
    tree=wrapper.toJSON()

    expect((tree as any).children[1].children[0]).toBe('1');
  })
  
  
});
