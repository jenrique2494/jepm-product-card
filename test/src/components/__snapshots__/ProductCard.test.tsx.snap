// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el ProductCard debe de incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Poduct card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`pruebas en el ProductCard debe mostrar el componente correctamente con el productcard personalizado 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    'hola mundo'
  </h1>
</div>
`;
